### DISCUSSION TITLE
Testing pyalarmdotcomajax v0.6.0-beta

### DISCUSSION THREAD

**Discussion by elahd** [pyalarmdotcom/alarmdotcom#391] [Show and tell]
I finished work on pyalarmdotcomajax v0.6.0-beta. This is a ground-up rebuild that's event driven and more robust than the current version. I'm looking for help testing to make sure that things work as expected before I start work on upgrading the Home Assistant integration.

### Installation & Use

This version of pyalarmdotcomajax needs a minimum Python version of 3.11. 

v0.6.0-beta is in the [refactor-2024 branch](https://github.com/pyalarmdotcom/pyalarmdotcomajax/tree/refactor-2024). This version can be installed with pip using `pip install git+https://github.com/pyalarmdotcom/pyalarmdotcomajax.git@refactor-2024`. This will give you access to the updated adc CLI tool, documentation [here](https://github.com/pyalarmdotcom/pyalarmdotcomajax/tree/refactor-2024/pyalarmdotcomajax/adc). 

`adc` has three top level commands: `get`, `stream`, and `action`.

1. `get` prints out a snapshot of your devices and their attributes. Use `adc get --help` for detailed help. Example usage: `adc get -u "USERNAME" -p "PASSWORD"`
2. `stream` monitors your system for real time changes and prints updated device data to the terminal when changes are detected. Use `adc get --help` for detailed help. Example usage: `adc stream -u "USERNAME" -p "PASSWORD"`
3. `action` lets you change the state of your devices -- you can take all actions available through the HA integration. For a list of all device types that can be manipulated, run `adc action --help`. For a list of all commands available for a device, run `adc action DEVICE_TYPE --help`. For a list of all command parameters, run `adc action DEVICE_TYPE COMMAND --help`. Example usage: `adc action light turn_on -u "USERNAME" -p "PASSWORD" DEVICE_ID`

### Additional Parameters & Functions

1. If you have a two factor authentication _cookie_, add it after the password argument as `-c "COOKIE"`. If you don't have a cookie but have MFA enabled, you'll be prompted for an OTP.
2. Add the `-d` flag to show debug output.
3. Add the `--json` flag to output raw device JSON instead of "pretty" tables.
4. To see raw data for all devices, including those not supported by the integration, run `get` with the `-x` flag.

### Goals

Generally, just kick the tires. Specifically, check that:

1. `get` returns a complete and accurate representation of your system's supported devices. (A list of supported device types is in the main [README](https://github.com/pyalarmdotcom/pyalarmdotcomajax/tree/refactor-2024).)
2. `stream` captures events for supported devices, doesn't disconnect excessively, and is able to successfully recover if disconnected.
3. `action` commands work as expected.

### Known Issues

1. The `stream` command automatically reconnects if it looses its connection, but it will not recover events lost while reconnected.
2. Contact sensors still follow the the 3-minute rule, meaning that they will not report the same state within a 3 minute time span. This means that you will only be notified once in the `stream` function if, say, a door has been opened and closed multiple times within 3 minutes. This will never be fixed as the limitation is enforced on Alarm.com's servers.
3. Some device types do not assign names to their entities. These entities will show up as "Unnamed DEVICE_TYPE". These types include device catalogs, image sensor images, and possibly a few others.

## Do you own a Skybell?

I dropped support for Skybells in v0.6.0 due to loss of access to a working device. Please post here if you're able to share some data from your Skybell to help keep support in the library.

🚀 Thanks!

**Comment by pdobrien3** [comment]
just checking in. Has anyone tried the beta? I have tried to follow the install instructions a couple times and never made it to completion.  Might be above my skill level.  Any chance of just building a beta release?
